source/common/thread/win32/thread_impl.cc

```diff
@@ -0,0 +1,28 @@
+#include "common/common/assert.h"
+#include "common/thread/thread_impl.h"
+
+namespace Envoy {
+namespace Thread {
+
+ThreadImpl::ThreadImpl(std::function<void()> thread_routine) : thread_routine_(thread_routine) {
+  RELEASE_ASSERT(Logger::Registry::initialized(), "");
+  thread_handle_ = ::CreateThread(NULL, 0,
```

Comment:
drive-by question: should this be _beginthreadex?  (from the [docs](https://docs.microsoft.com/en-us/windows/desktop/api/processthreadsapi/nf-processthreadsapi-createthread): `A thread in an executable that calls the C run-time library (CRT) should use the _beginthreadex` )